### PR TITLE
hotfix for boptest in threshold

### LINKED ISSUE #1260: OM Simulation Error with IDEAS.Examples.IBPSA.SingleZoneResidentialHydronicHeatPump
I'm hoping this helps with in getting IDEAS supported by OM, and will maybe show up in tests started with https://github.com/open-ideas/IDEAS/issues/1254.

OM Version: 1.20.0~dev-13-g85ae74c, Nightly build Ubuntu 18.04 x64
MSL Version: 4.0.0 distributed as OMEdit system library
IDEAS Version: master commit ad502f4f1d4160ddad6492036f3766362cedf49a
Model Path: IDEAS.Examples.IBPSA.SingleZoneResidentialHydronicHeatPump

In OMEdit, simulation error with both Dassl and Euler (tol 1e-6, start time = 0, final time = 864000) right after a seemingly successful initialization:

```
Process crashed Simulation process failed. Exited with code 11.
/tmp/OpenModelica_dhbubu18/OMEdit/IDEAS.Examples.IBPSA.SingleZoneResidentialHydronicHeatPump/SingleZoneResidentialHydronicHeatPump -port=45317 -logFormat=xmltcp -override=startTime=0,stopTime=864000,stepSize=30,tolerance=1e-6,solver=dassl,outputFormat=mat,variableFilter=.* -r=/tmp/OpenModelica_dhbubu18/OMEdit/IDEAS.Examples.IBPSA.SingleZoneResidentialHydronicHeatPump/SingleZoneResidentialHydronicHeatPump_res.mat -w -lv=LOG_STATS -inputPath=/tmp/OpenModelica_dhbubu18/OMEdit/IDEAS.Examples.IBPSA.SingleZoneResidentialHydronicHeatPump -outputPath=/tmp/OpenModelica_dhbubu18/OMEdit/IDEAS.Examples.IBPSA.SingleZoneResidentialHydronicHeatPump
The following assertion has been violated during initialization at time 0.000000
pum.nominalValuesDefineDefaultPressureCurve
*** Warning: You are using a flow or pressure controlled mover with the
             default pressure curve.
             This leads to approximate calculations of the electrical power
             consumption. Add the correct pressure curve in the record per
             to obtain an accurate computation.
             Setting nominalValuesDefineDefaultPressureCurve=true will suppress this warning.
The following assertion has been violated during initialization at time 0.000000
fan.nominalValuesDefineDefaultPressureCurve
*** Warning: You are using a flow or pressure controlled mover with the
             default pressure curve.
             This leads to approximate calculations of the electrical power
             consumption. Add the correct pressure curve in the record per
             to obtain an accurate computation.
             Setting nominalValuesDefineDefaultPressureCurve=true will suppress this warning.
The initialization finished successfully with 3 homotopy steps.
division by zero at time 0, (a=-0) / (b=-0), where divisor b expression is: -0.0
Process crashed
```

### PR DESCRIPTION
**Pull request by Mathadon**
for #1260

@dhblum does this work too? I used 1e-8 instead of 0.5 for the threshold.

@JavierArroyoBastida do you agree with this change?

closes #1260

### CODE REVIEW

**Comment by JavierArroyoBastida** [comment]
This is fine by me. Nice to see that you worked out https://github.com/open-ideas/IDEAS/issues/1260 and could nail down the problem. @Mathadon I think you mean 1e-8 instead of 0 (not 0.5)? I'd suggest documenting the motivation to use 1e-8, at least with a comment in the `const` block. Note also that BOPTEST will need to be updated to reflect these changes in the emulator.

**Comment by Mathadon** [comment]
So I mean 0.5, since that's what Dave suggested. I'll add documentation.

File: IDEAS/Examples/IBPSA/SingleZoneResidentialHydronicHeatPump.mo [hunk 1]
```diff
@@ -260,7 +260,7 @@ model SingleZoneResidentialHydronicHeatPump
     annotation (Placement(transformation(extent={{-80,90},{-60,70}})));
   Modelica.Blocks.Logical.Switch switch1(y(unit="K"))
     annotation (Placement(transformation(extent={{-20,140},{0,160}})));
-  Modelica.Blocks.Sources.Constant const(k=0)
+  Modelica.Blocks.Sources.Constant const(k=1e-8)
```
Comment (Mathadon) [inline IDEAS/Examples/IBPSA/SingleZoneResidentialHydronicHeatPump.mo:263]:
```suggestion
  Modelica.Blocks.Sources.Constant const(k=1e-8) "Threshold strictly larger than 0 for detecting occupancy"
```

**Comment by dhblum** [comment]
1e-8 is fine, 0.5 was just a course trial.  See https://github.com/open-ideas/IDEAS/pull/1264 though.

**Comment by dhblum** [comment]
Also @JavierArroyoBastida we can't update BOPTEST yet because this uses MSL4 and we don't yet have the tooling in BOPTEST parser to use OM.

**Pull request merged**